hvac/api/secrets_engines/transit.py

```diff
@@ -873,13 +874,18 @@ def sign_data(
                     "signature_algorithm": signature_algorithm,
                     "marshaling_algorithm": marshaling_algorithm,
                     "salt_length": salt_length,
+                    "batch_input": batch_input,
                 }
             )
         )
+
+        hash_algorithm = "" if not hash_algorithm else hash_algorithm
+
         api_path = utils.format_url(
-            "/v1/{mount_point}/sign/{name}",
+            "/v1/{mount_point}/sign/{name}/{hash_algorithm}",
```

Comment:
The description for this parameter says that it can _also_ be specified in the URL, not that it must be. The method seems to already submit it in parameters, so I'm not certain why we would require it to be in the URL. Let's continue this in the issue report: 
- https://github.com/hvac/hvac/issues/989

If you can update the tests in a way that demonstrates the issue, it would also be very helpful.

Reply:
> The description for this parameter says that it can _also_ be specified in the URL, not that it must be. The method seems to already submit it in parameters, so I'm not certain why we would require it to be in the URL. Let's continue this in the issue report:
> 
> * [transit sign_data doesn't take the hash_algorithm in consideration #989](https://github.com/hvac/hvac/issues/989)
> 
> If you can update the tests in a way that demonstrates the issue, it would also be very helpful.

hi @briantist , that's true but I got this issue when I was working on a project that needed the hash algorithm to be passed in the URL and after looking at Vault documentation here https://developer.hashicorp.com/vault/api-docs/secret/transit#hash-data I noticed that the hash algorithm is passed in the encryption URL. The changes I made here worked for my case.

Reply:
Let's continue this in the issue report. If you prefer, you can also change this PR to only add `batch_input` and then a separate PR could be used for the other thing once we figure out what's going on, or we can work that out first, whatever works for you.

Reply:
@briantist let's change this PR to `batch_input` for now.

Reply:
Since we are changing this PR to be for `batch_input` only, please undo these `hash_algorithm` changes for now